lib/FusionInventory/Agent/Task/WakeOnLan.pm

```diff
@@ -151,23 +153,37 @@ sub _getInterface {
             @interfaces = FusionInventory::Agent::Tools::Win32::getInterfaces(
                 logger => $self->{logger}
             );
+            # on Windows, we have to use internal device name instead of litteral name
+            for my $interface ( @interfaces ) {
+                $interface->{DESCRIPTION} =
+                    $self->_getWin32InterfaceId($interface->{PNPDEVICEID})
+            }
+
             last;
         }
     }
 
+    my @nonloopbackordumbinterfaces =
+        grep { $_->{DESCRIPTION} ne 'lo' }
+        grep { $_->{IPADDRESS} }
+        grep { $_->{MACADDR} }
+        @interfaces;
+
+   return @nonloopbackordumbinterfaces;
+}
+
+sub _getInterface {
```

Comment:
I don't see why you're keeping that private method. If I missed something, can you explain or can you just also remove it ?